### ISSUE TITLE
RTSP streaming error: Malformed SDP line: i={contentIsMalformed=true, dataType=4}

### ISSUE THREAD

**Issue by draskori** [androidx/media#1087]
I am streaming from camera in local network (RTSP H.264)  using androidx.media3:media3-exoplayer-rtsp:1.1.1 and I get following error in logcat:

Playback error
```
   androidx.media3.exoplayer.ExoPlaybackException: Source error
       at androidx.media3.exoplayer.ExoPlayerImplInternal.handleIoException(ExoPlayerImplInternal.java:684)
       at androidx.media3.exoplayer.ExoPlayerImplInternal.handleMessage(ExoPlayerImplInternal.java:660)
       at android.os.Handler.dispatchMessage(Handler.java:98)
       at android.os.Looper.loop(Looper.java:154)
       at android.os.HandlerThread.run(HandlerThread.java:61)
   Caused by: java.io.IOException: androidx.media3.common.ParserException: Malformed SDP line: i={contentIsMalformed=true, dataType=4}
       at androidx.media3.exoplayer.rtsp.RtspMediaPeriod$InternalListener.onSessionTimelineRequestFailed(RtspMediaPeriod.java:689)
       at androidx.media3.exoplayer.rtsp.RtspClient.dispatchRtspError(RtspClient.java:322)
       at androidx.media3.exoplayer.rtsp.RtspClient.access$700(RtspClient.java:76)
       at androidx.media3.exoplayer.rtsp.RtspClient$MessageListener.handleRtspResponse(RtspClient.java:682)
       at androidx.media3.exoplayer.rtsp.RtspClient$MessageListener.handleRtspMessage(RtspClient.java:515)
       at androidx.media3.exoplayer.rtsp.RtspClient$MessageListener.lambda$onRtspMessageReceived$0$androidx-media3-exoplayer-rtsp-RtspClient$MessageListener(RtspClient.java:508)
       at androidx.media3.exoplayer.rtsp.RtspClient$MessageListener$$ExternalSyntheticLambda0.run(D8$$SyntheticClass)
       at android.os.Handler.handleCallback(Handler.java:751)
       at android.os.Handler.dispatchMessage(Handler.java:95)
       at android.os.Looper.loop(Looper.java:154) 
       at android.os.HandlerThread.run(HandlerThread.java:61) 
   Caused by: androidx.media3.exoplayer.rtsp.RtspMediaSource$RtspPlaybackException: androidx.media3.common.ParserException: Malformed SDP line: i={contentIsMalformed=true, dataType=4}
```

**Comment by microkatz** [comment]
@draskori 

It seems that your camera is providing a malformed SDP description for its stream. Its supplying an "i" attribute which is a "session description" without a value. The SDP description are attribute-value pairs and so it cannot be empty.

Maybe there is some mechanism in the camera setup through which you can "name" the stream or fix it so that its not empty?

**Comment by draskori** [comment]
Unfortunately there isn't such option in setup... But I can stream it with VLC or VideoView on some Android 7 for example

**Comment by microkatz** [comment]
@draskori 

I pushed a fix, commit https://github.com/androidx/media/commit/52c1d60d39f690b7f3231095fb193510c93fd1ec, to the androidx/media main branch that should address your issue. It will allow an session information attribute with an empty value.

Hopefully that fixes your issue!

@skipt1206 
As long as the unknown attribute letter contains a value, then ExoPlayer should ignore it. Are you running into an issue with custom SDP descriptions that ExoPlayer crashes on?

**Comment by skipt1206** [comment]
> @draskori
> 
> I pushed a fix, commit [52c1d60](https://github.com/androidx/media/commit/52c1d60d39f690b7f3231095fb193510c93fd1ec), to the androidx/media main branch that should address your issue. It will allow an session information attribute with an empty value.
> 
> Hopefully that fixes your issue!
> 
> @skipt1206 As long as the unknown attribute letter contains a value, then ExoPlayer should ignore it. Are you running into an issue with custom SDP descriptions that ExoPlayer crashes on?

Thanks Michael.  I am not a developer, but I have some experience working with developers.  In this case I am trying to coordinate resolution of a problem getting Reolink camera rtsp to work with an app called Sparkle TV.  Your fix should help a lot.

**Comment by microkatz** [comment]
As the commit has fixed the initial problem I will close this issue. Please feel free to open a new ticket if an issue still exists.
